### PR TITLE
feat(bench): add research bench creation script

### PR DESCRIPTION
**Pull request by Kunlun-Zhu**
Closes # 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed
- [x] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [x] Ready for code review

### CODE REVIEW

**Comment by Kunlun-Zhu** [comment]
will add evaluation pipeline after this branch merged

File: research_bench/scripts/create_crossbench.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+def extract_arxiv_id_from_url(url: str) -> Optional[str]:
+    """
+    从 arXiv 链接中提取 arXiv ID。
+
+    Args:
+        url (str): arXiv 论文链接，例如 "https://arxiv.org/abs/2402.13448"
+
+    Returns:
+        str 或 None: 提取的 arXiv ID 或者 None（如果提取失败）
```
Comment (lwaekfjlk) [inline research_bench/scripts/create_crossbench.py:19]: delet all the chinese comments

File: research_bench/benchmark/cross_bench.json
Comment (lwaekfjlk) [inline research_bench/benchmark/cross_bench.json:1]: this file should be moved to huggingface.

**Review by lwaekfjlk — CHANGES_REQUESTED** [review]

File: research_bench/scripts/arxiv_link.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+https://arxiv.org/abs/2402.13448
+https://arxiv.org/abs/2409.19100
```
Comment (lwaekfjlk) [inline research_bench/scripts/arxiv_link.txt:2]: this txt is for cross-bench? if so, maybe put it somewhere else or delete?

**Review by lwaekfjlk — CHANGES_REQUESTED** [review]

File: research_bench/scripts/create_mlbench.py [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+            papers.append(paper)
+            existing_arxiv_ids.add(paper.get_short_id())
+            if len(papers) >= max_papers * 5:
+                break
+
+    logging.info(f"Fetched {len(papers)} papers for keyword: '{keyword}'")
+    return papers
+
+
+def fetch_references(
```
Comment (lwaekfjlk) [inline research_bench/scripts/create_mlbench.py:92]: are these functions very useful? if so, integrate them into the research town utils?

Comment (lwaekfjlk) [reply research_bench/scripts/create_mlbench.py:92]: this can be useful to put in the paper_collector.py

**Review by lwaekfjlk — CHANGES_REQUESTED** [review]

File: research_bench/scripts/arxiv_link.txt (hunk 2 above)
Comment (Kunlun-Zhu) [reply research_bench/scripts/arxiv_link.txt:2]: Those are used in the cross-domain generation

Comment (Kunlun-Zhu) [reply research_bench/scripts/arxiv_link.txt:2]: Where do you want to put?

File: research_bench/scripts/create_mlbench.py (hunk 3 above)
Comment (Kunlun-Zhu) [reply research_bench/scripts/create_mlbench.py:92]: Maybe another issue

File: research_bench/scripts/arxiv_link.txt (hunk 2 above)
Comment (lwaekfjlk) [reply research_bench/scripts/arxiv_link.txt:2]: maybe delete temporarily? These data are included in the crossbench.json, right?

File: research_bench/scripts/create_mlbench.py (hunk 3 above)
Comment (lwaekfjlk) [reply research_bench/scripts/create_mlbench.py:92]: ok,but make sure the comments are english.

**Review by lwaekfjlk — APPROVED** [review]

**Review by lwaekfjlk — APPROVED** [review]

**Pull request merged**